MultiClockCoreplex does not work after merging #502

I do the following modification after merging #502 
```diff
diff --git a/src/main/scala/rocketchip/RocketPlexMaster.scala b/src/main/scala/rocketchip/RocketPlexMaster.scala
index 05a4c33..854cb71 100644
--- a/src/main/scala/rocketchip/RocketPlexMaster.scala
+++ b/src/main/scala/rocketchip/RocketPlexMaster.scala
@@ -14,7 +14,8 @@ trait RocketPlexMaster extends L2Crossbar {
   val module: RocketPlexMasterModule
   val mem: Seq[TLInwardNode]

-  val coreplex = LazyModule(new DefaultCoreplex)
+//  val coreplex = LazyModule(new DefaultCoreplex)
+  val coreplex = LazyModule(new MultiClockCoreplex)

   coreplex.l2in := l2.node
   socBus.node := coreplex.mmio
```

But it reports the following errors while generating the fir code
```
Exception in thread "main" firrtl.passes.PassExceptions:
firrtl.passes.CheckChirrtl$UndeclaredReferenceException:  @[Tile.scala 59:66:rocketchip.DefaultConfig.fir@134508.4]: [module AsyncRocketTile] Reference _T_799 is not declared.
firrtl.passes.CheckChirrtl$UndeclaredReferenceException:  @[RocketTiles.scala 59:65:rocketchip.DefaultConfig.fir@143692.4]: [module MultiClockCoreplex_coreplex] Reference _T_799 is not declared.
firrtl.passes.PassException: 2 errors detected!
        at firrtl.passes.Errors.trigger(Passes.scala:29)
        at firrtl.passes.CheckChirrtl$.run(CheckChirrtl.scala:126)
        at firrtl.SimpleRun$$anonfun$runPasses$1$$anonfun$1.apply(Compiler.scala:107)
        at firrtl.SimpleRun$$anonfun$runPasses$1$$anonfun$1.apply(Compiler.scala:107)
        at firrtl.Utils$.time(Utils.scala:23)
        at firrtl.SimpleRun$$anonfun$runPasses$1.apply(Compiler.scala:107)
        at firrtl.SimpleRun$$anonfun$runPasses$1.apply(Compiler.scala:106)
        at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
        at scala.collection.immutable.List.foldLeft(List.scala:84)
        at firrtl.SimpleRun$class.runPasses(Compiler.scala:106)
        at firrtl.PassBasedTransform.runPasses(Compiler.scala:124)
        at firrtl.PassBased$class.runPasses(Compiler.scala:120)
        at firrtl.PassBasedTransform.runPasses(Compiler.scala:124)
        at firrtl.PassBasedTransform.execute(Compiler.scala:128)
        at firrtl.Compiler$$anonfun$4$$anonfun$5.apply(Compiler.scala:243)
        at firrtl.Compiler$$anonfun$4$$anonfun$5.apply(Compiler.scala:243)
        at firrtl.Utils$.time(Utils.scala:23)
        at firrtl.Compiler$$anonfun$4.apply(Compiler.scala:243)
        at firrtl.Compiler$$anonfun$4.apply(Compiler.scala:242)
        at scala.collection.LinearSeqOptimized$class.foldLeft(LinearSeqOptimized.scala:124)
        at scala.collection.immutable.List.foldLeft(List.scala:84)
        at firrtl.Compiler$class.compile(Compiler.scala:242)
        at firrtl.VerilogCompiler.compile(LoweringCompilers.scala:128)
        at firrtl.Driver$.execute(Driver.scala:116)
        at firrtl.Driver$.execute(Driver.scala:143)
        at firrtl.Driver$.main(Driver.scala:158)
        at firrtl.Driver.main(Driver.scala)
make: *** [/home/yuzihao/riscv-pard/rocket-chip/emulator/generated-src/rocketchip.DefaultConfig.v] Error 1
```

The above errors do not occur when I checkout to the commit before merging #502. So there may be something wrong to use MultiClockCoreplex in #502.

--- a/regression/Makefile
+++ b/regression/Makefile
@@ -58,7 +58,7 @@ endif
 
 ifeq ($(SUITE),GroundtestSuiteA)
 PROJECT=groundtest
-CONFIGS=MemtestConfig MemtestBufferlessConfig MemtestStatelessConfig FancyMemtestConfig
+CONFIGS=MemtestConfig MemtestBufferlessConfig MemtestStatelessConfig
 endif
 
 ifeq ($(SUITE),GroundtestSuiteB)

--- a/src/main/scala/coreplex/Configs.scala
+++ b/src/main/scala/coreplex/Configs.scala
@@ -77,7 +77,7 @@ class BaseCoreplexConfig extends Config ((site, here, up) => {
 })
 
 class WithNCores(n: Int) extends Config((site, here, up) => {
-  case NTiles => n
+  case RocketConfigs => List.fill(n){ RocketConfig(site(XLen)) }
 })
 
 class WithNBanksPerMemChannel(n: Int) extends Config((site, here, up) => {

--- a/src/main/scala/coreplex/RocketTiles.scala
+++ b/src/main/scala/coreplex/RocketTiles.scala
@@ -49,6 +49,7 @@ trait HasSynchronousRocketTilesModule extends CoreplexRISCVPlatformModule {
 trait HasAsynchronousRocketTiles extends CoreplexRISCVPlatform {
   val module: HasAsynchronousRocketTilesModule
 
+  import rocket.AsyncRocketTile
   val rocketTiles: Seq[AsyncRocketTile] = p(RocketConfigs).map { c =>
     LazyModule(new AsyncRocketTile(c)(p.alterPartial {
       case SharedMemoryTLEdge => l1tol2.node.edgesIn(0)

--- a/src/main/scala/rocket/Rocc.scala
+++ b/src/main/scala/rocket/Rocc.scala
@@ -75,7 +75,7 @@ trait CanHaveLegacyRoccsModule extends CanHaveSharedFPUModule with CanHavePTWMod
 
 }
 
-class LegacyRoccComplex(implicit p: Parameters) extends LazyModule with HasCoreParameters {
+class LegacyRoccComplex(implicit p: Parameters) extends LazyModule {
   val buildRocc = p(BuildRoCC)
   val usingRocc = !buildRocc.isEmpty
   val nRocc = buildRocc.size
@@ -84,11 +84,11 @@ class LegacyRoccComplex(implicit p: Parameters) extends LazyModule with HasCoreP
   val nPTWPorts = buildRocc.map(_.nPTWPorts).sum
   val roccOpcodes = buildRocc.map(_.opcodes)
 
-  val legacies = List.fill(nMemChannels) { LazyModule(new TLLegacy) }
+  val legacies = List.fill(nMemChannels) { LazyModule(new TLLegacy()(p.alterPartial({ case PAddrBits => 32 }))) }
   val masterNodes = legacies.map(_ => TLOutputNode())
   legacies.zip(masterNodes).foreach { case(l,m) => m := TLHintHandler()(l.node) }
 
-  lazy val module = new LazyModuleImp(this) {
+  lazy val module = new LazyModuleImp(this) with HasCoreParameters {
     val io = new Bundle {
       val tl = masterNodes.map(_.bundleOut)
       val dcache = Vec(nRocc, new HellaCacheIO)
@@ -117,7 +117,7 @@ class LegacyRoccComplex(implicit p: Parameters) extends LazyModule with HasCoreP
         case RoccNMemChannels => accelParams.nMemChannels
         case RoccNPTWPorts => accelParams.nPTWPorts
       }))
-      val dcIF = Module(new SimpleHellaCacheIF)
+      val dcIF = Module(new SimpleHellaCacheIF()(p.alterPartial({ case CacheName => CacheName("L1D") })))
       rocc.io.cmd <> cmdRouter.io.out(i)
       rocc.io.exception := io.core.exception
       dcIF.io.requestor <> rocc.io.mem
@@ -127,7 +127,7 @@ class LegacyRoccComplex(implicit p: Parameters) extends LazyModule with HasCoreP
       rocc
     }
 
-    (nRocc to legacies.size) zip roccs.map(_.io.utl) foreach { case(i, utl) =>
+    (nRocc until legacies.size) zip roccs.map(_.io.utl) foreach { case(i, utl) =>
       legacies(i).module.io.legacy <> utl
     }
     io.core.busy := cmdRouter.io.busy || roccs.map(_.io.busy).reduce(_ || _)

--- a/src/main/scala/rocket/Tile.scala
+++ b/src/main/scala/rocket/Tile.scala
@@ -61,7 +61,7 @@ class AsyncRocketTile(c: RocketConfig)(implicit p: Parameters) extends LazyModul
 
   lazy val module = new LazyModuleImp(this) {
     val io = new Bundle {
-      val master = masterNodes.map(_.bundleOut)
+      val master = masterNodes.head.bundleOut // TODO fix after Chisel #366
       val slave = slaveNode.map(_.bundleIn)
       val hartid = UInt(INPUT, p(XLen))
       val interrupts = new TileInterrupts()(p).asInput

